File: Leetcode/LC33 Search in Rotated Sorted Array.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        if(nums[0] == target)
            return 0;
        else if(nums.size() == 1)
            return -1;
        int count;
        if(nums[0] > target){
            count = nums.size()-1;
            while(nums[count] != target){
                if(nums[count] < target )
                    return -1;
                --count;
                if(count <0)
                    return -1;
            }
        }
        else{
            count = 1;
            while( nums[count] != target){
                if(nums[count] > target )
                    return -1;
                ++count;
                if(count >= nums.size())
                    return -1;
            }
        }
        return count;
       
    }
};

File: Leetcode/LC33 Search in Rotated Sorted Array.ts
function search(nums: number[], target: number): number {
    let l = 0,r = nums.length-1;
    if(r===0 && nums[r] === target)
        return 0;
    while(nums[l]>nums[r]){ // find pivot
        let p = Math.floor((l+r)/2);
        if(nums[l] < nums[p]){
            l=p;
        }
        else{
            r = p;
        }
    }
    let pet = (r+1)%nums.length;
    l = 0,r = nums.length-1;
    while(l<=r){
       

        let p = Math.floor((l+r)/2) + pet;
        p%=nums.length;
        if(target > nums[p] ){
            l = Math.floor((l+r)/2)+1;
        }
        else if(target < nums[p]){
            r = Math.floor((l+r)/2)-1;
        }
        else{
            return p;
        }
    }
    return -1;
};